src/Persistence_matrix/concept/PersistenceMatrixColumn.h

```diff
@@ -0,0 +1,470 @@
+   * @ref Cell_field_element::get_element "Cell::get_element".
+   * @return Reference to this column.
+   */
+  template <class Cell_range>
+  PersistenceMatrixColumn& multiply_and_add(const Field_element_type& val, const Cell_range& column);
+  /**
+   * @brief `this = val * this + column`
+   * 
+   * @param val Value to multiply.
+   * @param column Column to add.
+   * @return Reference to this column.
+   */
+  PersistenceMatrixColumn& multiply_and_add(const Field_element_type& val, PersistenceMatrixColumn& column);
+  /**
+   * @brief `this = this + column * val`
+   * 
+   * @tparam Cell_range @ref Cell range with %begin() and %end() method.
+   * @param column @ref Cell range. Every cell has to return the right value when using @ref Cell::get_row_index and,
+   * if @ref PersistenceMatrixOptions::is_z2 is false, also when using
+   * @ref Cell_field_element::get_element "Cell::get_element".
+   * @param val Value to multiply.
+   * @return Reference to this column.
+   */
+  template <class Cell_range>
+  PersistenceMatrixColumn& multiply_and_add(const Cell_range& column, const Field_element_type& val);
+  /**
+   * @brief `this = this + column * val`
+   * 
+   * @param column Column to add.
+   * @param val Value to multiply.
+   * @return Reference to this column.
+   */
+  PersistenceMatrixColumn& multiply_and_add(PersistenceMatrixColumn& column, const Field_element_type& val);
+
+  /**
+   * @brief Equality comparator. Equal in the sense that what is "supposed" to be contained in the columns is equal,
+   * not what is actually stored in the underlying container. For exemple, the underlying container of 
+   * @ref Vector_column can contain cells which were erased explicitely by @ref clear(index). Those cells should not
+   * be taken into account while comparing.
+   * 
+   * @param c1 First column to compare.
+   * @param c2 Second column to compare.
+   * @return true If both column are equal.
+   * @return false Otherwise.
+   */
+  friend bool operator==(const PersistenceMatrixColumn& c1, const PersistenceMatrixColumn& c2);
+  /**
+   * @brief "Strictly smaller than" comparator. Usually a lexicographical order, but what matters is that the
+   * order is total. The order should apply on what is "supposed" to be contained in the columns,
+   * not what is actually stored in the underlying container. For exemple, the underlying container of 
+   * @ref Vector_column can contain cells which were erased explicitely by @ref clear(index). Those cells should not
+   * be taken into account while comparing.
+   * 
+   * @param c1 First column to compare.
+   * @param c2 Second column to compare.
+   * @return true If the first column is strictly smaller than the second one.
+   * @return false Otherwise.
+   */
+  friend bool operator<(const PersistenceMatrixColumn& c1, const PersistenceMatrixColumn& c2);
+
+  /**
+   * @brief Assign operator. Should be disabled when row access is enabled.
```

Comment:
I debated with myself a lot about this. The problem is that if we allow assignment with row access, it is not clear at all, what to do with the rows. Should the column take over the pointer to the rows from the other or not? What if I just want a dummy column outside of the matrix and therefore I don't want to waste time on the rows? I wanted to see how the columns are used in the first place before implementing something for it. I enabled it for the other case, because it ended being quite useful in this particular case.

Reply:
Could you maybe point out an example or 2 where you use it in the code? IIUC it has to be in code that is never used (doesn't make sense?) with columns that have row access, and I wonder what kind of code that is. Just the compressed annotation matrix?

Reply:
I used it in situation where it would also be useful for row access, but I couldn't decide on a single politic for this case, so I just did it for no row access and changed strategy for row access. For example, when pre-allocating a vector with dummy colums and then assign a real column to each vector cell afterwards. Typically, I cannot do that with row access, which is a bit annoying.